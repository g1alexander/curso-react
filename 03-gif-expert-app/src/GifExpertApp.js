import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export function GifExpertApp({ defaultCategories = [] }) {
  const [categories, setCategories] = useState(defaultCategories);

  // const handleAdd = () => {
  //   setCategories([...categories, "five"]);
  //    setCategories(cat => [...cat, "five"]);
  // };

  return (
    <>
      <h1>GifExpertApp</h1>
      <hr />

      <AddCategory setCategories={setCategories} />

      {/* <button onClick={handleAdd}>Agregar</button> */}
      <ol>
        {categories.map((category) => (
          <GifGrid category={category} key={category} />
        ))}
      </ol>
    </>
  );
}
