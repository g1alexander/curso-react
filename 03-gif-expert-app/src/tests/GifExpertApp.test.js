import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("pruebas en el componente <GifExpertApp />", () => {
  test("se debe mostrar correctamente el componente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrarse un item cuando se le pasa una categoria", () => {
    const categories = ["the beatles"];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
