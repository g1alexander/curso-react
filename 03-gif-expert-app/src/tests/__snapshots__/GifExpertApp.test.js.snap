// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifExpertApp /> debe mostrarse un item cuando se le pasa una categoria 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="the beatles"
      key="the beatles"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en el componente <GifExpertApp /> se debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
